internal/services/registry/client.go

```diff
@@ -75,6 +75,82 @@ func (m *MintVehicleSign) Message() signer.TypedDataMessage {
 	}
 }
 
+// MintVehicleWithDeviceDefinitionSign(uint256 manufacturerNode, address owner, string deviceDefinitionId, (string,string)[] attrInfo, bytes signature)
+type MintVehicleWithDeviceDefinitionSign struct {
+	ManufacturerNode   *big.Int
+	Owner              common.Address
+	Attributes         []string
+	Infos              []string
+	DeviceDefinitionID string
+}
+
+func (m *MintVehicleWithDeviceDefinitionSign) Name() string {
+	return "MintVehicleWithDeviceDefinitionSign"
+}
+
+func (m *MintVehicleWithDeviceDefinitionSign) Type() []signer.Type {
+	return []signer.Type{
+		{Name: "manufacturerNode", Type: "uint256"},
+		{Name: "owner", Type: "address"},
+		{Name: "attributes", Type: "string[]"},
+		{Name: "infos", Type: "string[]"},
+		{Name: "deviceDefinitionId", Type: "string"},
+	}
+}
+
+func (m *MintVehicleWithDeviceDefinitionSign) Message() signer.TypedDataMessage {
+	return signer.TypedDataMessage{
+		"manufacturerNode":   hexutil.EncodeBig(m.ManufacturerNode),
+		"owner":              m.Owner.Hex(),
+		"attributes":         anySlice(m.Attributes),
+		"infos":              anySlice(m.Infos),
+		"deviceDefinitionId": m.DeviceDefinitionID,
+	}
+}
+
+// MintVehicleAndSdSign(uint256 integrationNode)
+// Only signed by the synthetic device's wallet.
+type MintVehicleAndSdSign struct {
+	IntegrationNode *big.Int
+}
+
+func (m *MintVehicleAndSdSign) Name() string {
+	return "MintVehicleAndSdSign"
+}
+
+func (m *MintVehicleAndSdSign) Type() []signer.Type {
+	return []signer.Type{
+		{Name: "integrationNode", Type: "uint256"},
+	}
+}
+
+func (m *MintVehicleAndSdSign) Message() signer.TypedDataMessage {
+	return signer.TypedDataMessage{
+		"integrationNode": hexutil.EncodeBig(m.IntegrationNode),
+	}
+}
+
+// MintVehicleAndSdWithDeviceDefinitionSign(uint256 integrationNode)
+type MintVehicleAndSdWithDeviceDefinitionSign struct {
+	IntegrationNode *big.Int
+}
+
+func (m *MintVehicleAndSdWithDeviceDefinitionSign) Name() string {
+	return "MintVehicleAndSdWithDeviceDefinitionSign"
```

Comment:
I don't think this is a real entity in dimo-identity.

Reply:
Like in both cases, this is what's used https://github.com/DIMO-Network/dimo-identity/blob/811a89081cc89c821cb0b37c2ac6796731af743d/contracts/implementations/MultipleMinter.sol#L28

Reply:
isn't that this (from the go file generated with the abi): https://github.com/DIMO-Network/devices-api/blob/main/internal/contracts/registry.go#L1942

Reply:
That's a function, not an EIP-712 payload. The payloads always look like what I linked.